platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/proof/StateProof.java

```diff
@@ -0,0 +1,192 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.merkle.proof;
+
+import static com.swirlds.common.merkle.proof.algorithms.StateProofSerialization.deserializeSignatures;
+import static com.swirlds.common.merkle.proof.algorithms.StateProofSerialization.deserializeStateProofTree;
+import static com.swirlds.common.merkle.proof.algorithms.StateProofSerialization.extractPayloads;
+import static com.swirlds.common.merkle.proof.algorithms.StateProofSerialization.serializeSignatures;
+import static com.swirlds.common.merkle.proof.algorithms.StateProofSerialization.serializeStateProofTree;
+import static com.swirlds.common.merkle.proof.algorithms.StateProofTreeBuilder.buildStateProofTree;
+import static com.swirlds.common.merkle.proof.algorithms.StateProofTreeBuilder.processSignatures;
+import static com.swirlds.common.merkle.proof.algorithms.StateProofTreeBuilder.validatePayloads;
+import static com.swirlds.common.merkle.proof.algorithms.StateProofValidator.computeStateProofTreeHash;
+import static com.swirlds.common.merkle.proof.algorithms.StateProofValidator.computeValidSignatureWeight;
+
+import com.swirlds.common.crypto.Cryptography;
+import com.swirlds.common.crypto.Signature;
+import com.swirlds.common.io.SelfSerializable;
+import com.swirlds.common.io.streams.SerializableDataInputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.common.merkle.MerkleLeaf;
+import com.swirlds.common.merkle.MerkleNode;
+import com.swirlds.common.merkle.proof.algorithms.NodeSignature;
+import com.swirlds.common.merkle.proof.algorithms.SignatureValidator;
+import com.swirlds.common.merkle.proof.tree.StateProofNode;
+import com.swirlds.common.system.NodeId;
+import com.swirlds.common.system.address.AddressBook;
+import com.swirlds.common.utility.Threshold;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A state proof on one or more merkle nodes.
+ * <p>
+ * Warning: this is an unstable API, and it may be changed and/or removed suddenly and without warning.
+ */
+public class StateProof implements SelfSerializable {
+
+    private static final long CLASS_ID = 0xbf5d45fc18b63224L;
+
+    private static final class ClassVersion {
+        public static final int ORIGINAL = 1;
+    }
+
+    private List<NodeSignature> signatures;
+    private StateProofNode root;
+    private List<MerkleLeaf> payloads;
```

Comment:
A think a comment explaining what a `payload` is would be good

Reply:
```
    /**
     * A payload is a leaf node in the merkle tree that is being proven. There will be exactly one payload for each
     * merkle leaf that is being proven.
     */
    private List<MerkleLeaf> payloads;
```